### PR TITLE
feat(auth): build token implementation

### PR DESCRIPTION
**Pull request by ecrupper**
Server PR for Epic 1: Build Tokens.

Changes:

- Conversion of `router/middleware/token` package to `router/middleware/auth`, which will now only retrieve auth headers from http requests, rather than generate/parse tokens
- Creation of `internal/token` package, which features a `manager` that is set up in the context using various new flags added to `main.go`. In brief, it's a one-stop-shop for token minting and validating and offers configurability to a degree.
- Additional claims added to `Claims` struct, including information about the token type as well as relevant fields for that token (build tokens have build ids and repo).
- Removal of global permissions for agents
- Worker endpoints used for check in (POST / PUT) have been put behind `MustWorker()` permissions check, which validates the symmetric token shared between server and worker
- Build/Step/Service/Log update endpoints have been put behind `MustBuildAccess()` permissions check, which validates the build token, which is provided as a unique auth scope for each build.
- Add GET endpoint for retrieving a build token, which validates the build's status and creates a build token
- Addition of a private key, which will be seeded in the server environment, much like the secret, which will be used to sign and validate all tokens.
- Updates made to `MustSecretAdmin` which will scope requests with build token auth based on the repo within the token
- Claims will be established in the context for baseAPI endpoints before the user

depends on: https://github.com/go-vela/types/pull/276

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -1,52 +0,0 @@
-# name of the action
```
Comment (plyr4) [inline .github/workflows/publish.yml:1]: revert deletion?

**Review by wass3r — APPROVED** [review]

**Review by wass3r — DISMISSED** [review]

File: internal/token/compose.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	"net/http"
+	"net/url"
+
+	"github.com/gin-gonic/gin"
+	"github.com/go-vela/types"
+	"github.com/go-vela/types/constants"
+	"github.com/go-vela/types/library"
+)
+
+// Compose generates an refresh and access token pair unique
```
Comment (plyr4) [inline internal/token/compose.go:17]:
```suggestion
// Compose generates a refresh and access token pair unique
```

**Review by plyr4 — CHANGES_REQUESTED** [review]

File: internal/token/compose.go (hunk 2 above)
Comment (ecrupper) [reply internal/token/compose.go:17]: Nice catch!!

**Review by wass3r — APPROVED** [review]

**Review by jbrockopp — CHANGES_REQUESTED** [review]
proposal?

**Comment by ecrupper** [comment]
> proposal?

https://github.com/go-vela/community/pull/755

**Review by wass3r — APPROVED** [review]

**Review by plyr4 — APPROVED** [review]

**Review by timhuynh94 — APPROVED** [review]

**Review by cognifloyd — APPROVED** [review]

**Pull request merged**